src/Orleans.Transactions/State/ReaderWriterLock.cs

```diff
@@ -274,108 +277,114 @@ public async Task Rollback(Guid guid, bool notify)
 
         private async Task LockWork()
         {
-            var now = DateTime.UtcNow;
-
-            if (currentGroup != null)
+            // Stop pumping lock work if this activation is stopping/stopped.
+            if (this.activationLifetime.OnDeactivating.IsCancellationRequested) return;
```

Comment:
Would it make sense to update the BatchWorker to take a cancelation token and stop scheduling work, rather than having each Work call need to implement this?